sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -155,6 +166,42 @@ class Filter(TransformStep[bool]):
     step_type: StepType = StepType.FILTER
 
 
+@dataclass
+class Branch(Step):
+    """
+    A Branch represents one branch in a pipeline, which is routed to
+    by a Router.
+    The name of the Branch step must match the key of the step in
+    its Router's routing table.
+    """
+
+    step_type: StepType = StepType.BRANCH
+
+
+@dataclass
+class Router(WithInput):
+    """
+    A step which takes a routing table of Branches and sends messages
+    to those branches based on a routing function.
+    Routing functions must only return a single output branch, routing
+    to multiple branches simultaneously is not currently supported.
+    """
+
+    routing_function: Callable[..., str]
```

Comment:
Asking to return a string is overly restrictive. People will want to use enums.
Could you please make this generic with respect to the return type of the routing_function. 

It should be possible as the router step or translator does not care in any way about the returned type. It only needs to map the returned type to a Branch.

In the same way `routing_table` should be generic with regards to this type.